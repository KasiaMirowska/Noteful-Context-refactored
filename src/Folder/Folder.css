.folder {
    flex-basis: 40%;
    color: rgb(28, 151, 28);
    background-color: white;
    font-family: 'Raleway', sans-serif;
    cursor: pointer;  
    margin:10px;
    right: 0;
    text-align: right;
}
.folder-list {
    width: 30%;
    margin-left: 60%;
    border: 1px solid lightgray; 
    text-align: right;
}