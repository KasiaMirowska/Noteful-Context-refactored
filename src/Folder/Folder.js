import React from 'react'
import './Folder.css';
import { Link } from 'react-router-dom';
import NotefulContext from '../NotefulContext';
import PropTypes from 'prop-types';


function deleteFolderRequest(folderId, callback) {
    const url = 'http://localhost:9090/folders'
    fetch(url + `/${folderId}`, {
        method: 'DELETE',
    })
    .then(res => {
        if(!res.ok) {
            return res.json().then(error => {
                throw error
            })
        }
        return res.json()
    })
    .then(data => {
        callback(folderId)
    })
    .catch(error => {
        console.log(error)
    })
}

export default function Folder(props) {
    return (
        <NotefulContext.Consumer>
            {(context) => (
        <div className='folder'>
            <ul className='folder-list'>
                <li key={props.id} >
                    <Link to= {`/folder/${props.id}`} className="link"> 
                    <h3>{props.name}</h3>
                    <p>{props.amount}</p>
                    </Link>
                    <button onClick={() => deleteFolderRequest(props.id, context.deleteFolder)}>Delete folder</button>
                </li>
            </ul>
        </div> 
            )
         }
        </NotefulContext.Consumer> 
)

    
}
Folder.propTypes = {
    folders: PropTypes.arrayOf(PropTypes.shape({
        name: PropTypes.string.isRequired,
        id: PropTypes.string.isRequired,
        amount: PropTypes.number,
    })),
    deleteFolder: PropTypes.func,
}
