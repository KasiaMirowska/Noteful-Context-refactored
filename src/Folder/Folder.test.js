import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Folder from './Folder';
import { BrowserRouter as BR} from 'react-router-dom';


describe('Folder component', () => {
    const props = {
        folders: [{
            name: 'some name',
            id: 'id',
        }]
    }
    it ('renders without crashing', () => {
        const div = document.createElement('div');

        ReactDOM.render(<BR><Folder {...props} /></BR>, div);
        ReactDOM.unmountComponentAtNode(div);
    })

    it('renders the UI as expected', ()=>{
        const folder = renderer.create(<BR><Folder {...props}/></BR>);

        expect(folder.toJSON()).toMatchSnapshot();
    })

    it('renders a clicked folder', () => {
    const wrapper = shallow(<BR><Folder match={{params: {folderId: 'id'}}}/></BR>)

        wrapper.find('Link').simulate('click');
        expect(wrapper.find({match:{path:'link'}}).exists()).toBe(true);
    })
})