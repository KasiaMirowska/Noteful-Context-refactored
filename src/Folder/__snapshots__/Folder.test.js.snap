// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Folder component renders the UI as expected 1`] = `
<div
  className="folder"
>
  <ul
    className="folder-list"
  >
    <li>
      <a
        className="link"
        href="/folder/undefined"
        onClick={[Function]}
      >
        <h3 />
      </a>
      <p>
        0
      </p>
      <button
        onClick={[Function]}
      >
        Delete folder
      </button>
    </li>
  </ul>
</div>
`;
